grammar.js

```diff
@@ -320,12 +320,11 @@ module.exports = grammar({
       'struct',
       $._type_identifier,
       optional($.type_parameters),
-      optional($.where_clause),
       choice(
-        $.field_declaration_list,
+        seq(optional($.where_clause), $.field_declaration_list),
         seq($.ordered_field_declaration_list, optional($.where_clause), ';'),
```

Comment:
Nice job tracking that problem down. I didn't fully understand the subtlety of how the `where` clause works differently for tuple structs than it does for structs with named fields.